sdk/src/Services/DynamoDBv2/Custom/DataModel/Configs.cs

```diff
@@ -317,6 +326,7 @@ public DynamoDBFlatConfig(DynamoDBOperationConfig operationConfig, DynamoDBConte
             List<ScanCondition> queryFilter = operationConfig.QueryFilter ?? new List<ScanCondition>();
             ConditionalOperatorValues conditionalOperator = operationConfig.ConditionalOperator;
             DynamoDBEntryConversion conversion = operationConfig.Conversion ?? contextConfig.Conversion ?? DynamoDBEntryConversion.CurrentConversion;
+            MetadataCachingMode metadataCachingMode = operationConfig.MetadataCachingMode ?? contextConfig.MetadataCachingMode ?? DynamoDBv2.MetadataCachingMode.Default;
```

Comment:
Added in https://github.com/aws/aws-sdk-net/compare/1161dc30f02a5a18108c28b0191f21da5080a368..0c2afdec6c32efc6ab9ebd3841a433bf19683181
 * I added a unit test that sets it via `AWSConfigsDynamoDB.Context.MetadataCachingMode`. 
 * I manually tested setting it via `app.config` in the Net35 and Net45 integration test projects, but did not check that in so that most of our existing tests will be seeing the null (implicit `Default`) behavior.